__tests__/index.test.js

```diff
@@ -71,7 +73,66 @@ describe("/api/articles/:article_id tests", () => {
         );
       });
   });
-  test("404: missing article id", () => {
-    return request(app).get("/api/articles/").expect(404);
+
+  test("200: sends an array of all articles if article id is not specified", () => {
+    return request(app).get("/api/articles/").expect(200);
+  });
+});
+
+describe("/api/articles tests", () => {
+  test("200: sends an array of articles to the client", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles.length).toBe(13);
+        expect(Array.isArray(response.body.articles)).toBeTruthy();
+      });
+  });
+  test("200: each article should have the properties of author, title, article_id, topic, created_at, votes, article_img_url and not have a body property", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        response.body.articles.forEach((article) => {
+          expect(article).toHaveProperty("author");
+          expect(article).toHaveProperty("title");
+          expect(article).toHaveProperty("article_id");
+          expect(article).toHaveProperty("topic");
+          expect(article).toHaveProperty("created_at");
+          expect(article).toHaveProperty("votes");
+          expect(article).toHaveProperty("article_img_url");
+          expect(article).not.toHaveProperty("body");
+        });
+      });
+  });
+  test("200: each article should have the comment_count, which is the total count of all the comments with this article_id.", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles[0].comment_count).toBe(2);
+      });
+  });
+  test("200: accepts a created_at query and sorts the articles by date in descending order.", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
+  test("404: incorrect endpoint", () => {
+    return request(app).get("/api/articules").expect(404);
+  });
```

Comment:
This test doesn't need to exist on each endpoint, we can test it once in a describe for example and it should cover all 404's